rfc/0015-multi-file-jayvee/0015-multi-file-jayvee.md

```diff
@@ -0,0 +1,160 @@
+
+**Access paths:**
+
+- root level elements can access
+  - root level elements by their name
+  - and elements of libraries by their qualified name
+- elements within a library can access
+  - root level elements by their name
+  - and elements of other libraries by their qualified name
+- elements within a pipeline can access
+  - root level elements by their name
+  - elements within the same pipeline by their name
+  - and elements of libraries by their qualified name
+
+**No-access paths:**
+
+- elements within a pipeline cannot be referenced by outside elements
+
+**Imported elements are handled as if they were defined at the root level.**
+
+### Importing elements
+
+Only `export`ed elements can be imported into other files.
+
+#### Importing exported elements of a file
+
+```
+from './path/to/location.jv' use { MyDomainSpecificValuetype1 }; // only imports the defined elements from the file, access via qualified name as if it would be defined at the root level
+from './path/to/location.jv' use { MyDomainSpecificValuetype1 called Vt1} // only imports the defined elements from the file, access via qualified name using the alias
+```
+
+References to these imported elements is by their qualified name (unless altered by an alias).
+
+#### Importing a library
+
+Each import explicitly defines the version of the imported library.
+On version mismatch, an error is raised.
+Libraries can only be imported as a whole.
+
+```
+from './path/to/location.jv' use { MyDomainLibrary version 1.2.3 }; // only imports the named library, access via qualified name
+from './path/to/location.jv' use {
+  MyDomainLibrary1 version 1.2.3,
+  MyDomainLibrary1 version 1.2.3
+}; // only imports the named libraries, access via qualified name
+from './path/to/location.jv' use { MyDomainLibrary version 1.2.3 called MyLibraryAlias} // only imports the named library, access via qualified name using the alias
+```
+
+References to these imported elements is by their qualified name (unless altered by an alias).
+
+## Drawbacks
+
+- Two different sharing mechanisms (export keyword, library)
+- Elements of a pipeline cannot be reused, leading to potentially more slim pipelines and a parallel library
+- The elements of a library within a file always need the qualified name (alternative: allow access via sole name within file?)
+- We do not allow re-exporting (only by putting elements into a containing library)
+- Langium might not support this scoping mechanism out-of-the-box (more complex implementation)
+
+## Alternatives
+
+- "use" syntax without braces, etc., `from './path/to/file.jv' use MyDomainLibrary1, MyDomainLibrary2`
+- different syntax for importing files and libraries
+- Rather call it `module` instead of `library`
+
+## Possible Future Changes/Enhancements
+
+- build out to use libraries of other projects via a package-manager mechanism, e.g., by using an URL as location of a "use" statement
```

Comment:
Ah, I'd have it in the RFC already but don't need to implement it at first.

Reply:
Suggestion: we define how importable locations look like (HTTP, file path) and leave the exact package-manager mechanism out of this PR (when to fetch remote libraries, where to cache them, etc.)